src/libs/actions/IOU.js

```diff
@@ -364,7 +377,7 @@ function requestMoney(report, amount, currency, payeeEmail, payeeAccountID, part
         receiptObject.source = receipt.source;
         receiptObject.state = CONST.IOU.RECEIPT_STATE.SCANREADY;
     }
-    const optimisticTransaction = TransactionUtils.buildOptimisticTransaction(amount, currency, iouReport.reportID, comment, '', '', undefined, receiptObject);
+    const transaction = TransactionUtils.buildOptimisticTransaction(amount, currency, iouReport.reportID, comment, '', '', undefined, receiptObject);
```

Comment:
For distance requests we create the transaction optimistically as soon as the flow starts, so we need to get that transaction and merge it with the other optimistic transaction data built here.
https://github.com/Expensify/App/blob/82633312c96994efaf664fbea4ed2bff598c986e/src/pages/iou/DistanceRequestPage.js#L29

Reply:
OK, it looks like `createEmptyTransaction()` is a method that generates a random transactionID (I think maybe it is a poorly named method). In order to use that transactionID, I have modified `createDistanceRequest()` to accept an existing transaction ID. This ID is then used when creating the optimistic transaction in this code instead of generating a new random ID.

I think that should be all that is needed for when someone hooks up to the `createDistanceRequest()` method, but please double-check my assumptions.

Reply:
We can build the optimistic transaction with the same ID, but then we still need to merge it with the transaction that already exists in Onyx. The existing transaction will have waypoints set on it, and maybe some other data that I don't think we want to lose.